# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentParent.h
# Commit: 30d5dfedf252
# Full Hash: 30d5dfedf252f4f10a0d8948e86f4e57f1c4ccf9
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-11 09:21:34
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 8 - CookieStore API - e10s support, r=smaug,cookie-reviewers,timhuang
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D217574
# ==============================================================================

diff -r fc3b4c5ee4b2 -r 30d5dfedf252 dom/ipc/ContentParent.h
--- a/dom/ipc/ContentParent.h	Tue Sep 10 16:31:40 2024 +0000
+++ b/dom/ipc/ContentParent.h	Tue Sep 10 16:31:41 2024 +0000
@@ -629,6 +629,12 @@
   // to this content process forever.
   void TransmitBlobURLsForPrincipal(nsIPrincipal* aPrincipal);
 
+  // Update a cache list of allowed domains to store cookies for the current
+  // process. This method is called when PCookieServiceParent actor is not
+  // available yet.
+  void AddPrincipalToCookieInProcessCache(nsIPrincipal* aPrincipal);
+  void TakeCookieInProcessCache(nsTArray<nsCOMPtr<nsIPrincipal>>& aList);
+
   nsresult TransmitPermissionsForPrincipal(nsIPrincipal* aPrincipal);
 
   // Whenever receiving a Principal we need to validate that Principal case
@@ -1569,6 +1575,8 @@
 
   nsTArray<nsCString> mBlobURLs;
 
+  nsTArray<nsCOMPtr<nsIPrincipal>> mCookieInContentListCache;
+
   // This is intended to be a memory and time efficient means of determining
   // whether an origin has ever existed in a process so that Blob URL broadcast
   // doesn't need to transmit every Blob URL to every content process. False